### ISSUE TITLE
macOS BSD tar: Option --warning=none is not supported

### ISSUE THREAD

**Issue by kiryph** [gap-packages/PackageManager#97]
On macOS Big Sur (11.6.8) the preinstalled `/usr/bin/tar` returns an error.

```
gap> LoadPackage("PackageManager");
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  PackageManager 1.3 (Easily download and install GAP packages)
by Michael Young (https://mct25.host.cs.st-andrews.ac.uk/).
Homepage: https://gap-packages.github.io/PackageManager/
Report issues at https://github.com/gap-packages/PackageManager/issues
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
true

gap> InstallPackage("curlInterface");
#I  Getting PackageInfo URLs...
#I  Retrieving PackageInfo.g from https://gap-packages.github.io/curlInterface/PackageInfo.g ...
#I  Downloading archive from URL https://github.com/gap-packages/curlInterface/releases/download/v2.2.3/curlInterface-2.2.3.tar.gz ...
#I  Saved archive to /var/folders/2z/bljkcy1j2cbf7l9zf3dk0nqc0000gn/T//gaptempdiriZYv5Y/curlInterface-2.2.3.tar.gz.pkgman
#I  Possible error detected: see log at /var/folders/2z/bljkcy1j2cbf7l9zf3dk0nqc0000gn/T//gaptempdirpvwRQF/exec-log.txt
#I  Could not inspect tarball contents
```

```
❯ cat /var/folders/2z/bljkcy1j2cbf7l9zf3dk0nqc0000gn/T//gaptempdirpvwRQF/exec-log.txt
tar: Option --warning=none is not supported
Usage:
  List:    tar -tf <archive-filename>
  Extract: tar -xf <archive-filename>
  Create:  tar -cf <archive-filename> [filenames...]
  Help:    tar --help

❯ tar --version
bsdtar 3.3.2 - libarchive 3.3.2 zlib/1.2.11 liblzma/5.0.5 bz2lib/1.0.6
```

Does the Package Manager really need GNU tar?

What about simply removing `--warning=none`?

When I install GNU tar on macOS with `$ brew install gnu-tar`, it will be installed as `gtar`. Is it possible to specify the tar binary for the PackageManager?

**Comment by kiryph** [comment]
Workaround install via git repo:
```
gap> InstallPackage("https://github.com/gap-packages/curlInterface.git");
#I  Cloning to /Users/kiryph/Library/Preferences/GAP/pkg/curlInterface ...
#I  Package cloned successfully
#I  Checking dependencies for curlInterface...
#I    GAPDoc >= 1.5: true
#I  Building documentation (using makedoc.g)...
true
```

**Comment by fingolfin** [comment]
That's a bug, introduced in PR #82 by @ChrisJefferson . It definitely should not pass that option to non-GNU-tar variants.

One way to fix this would be to detect whether `tar` is GNU tar or not, and only enable the workaround for issue #80 from PR #82 in that case.

However, I think the *real* problem here is that `PKGMAN_Exec` deliberately merges `stderr` and `stdout`, which breaks here. I have my doubts that this is a good idea in general; it definitely is not here. Perhaps `PKGMAN_Exec` could get an optional argument which can be used to turn this feature off.

I think it would be better to never merged stderr into stdout and instead redirect it into a log file.

**Comment by mtorpey** [comment]
@kiryph: Pull request #99 should hopefully have fixed this, though it might not have fixed the underlying issue.  Could you verify whether it's fixed things for you?

**Comment by fingolfin** [comment]
I also can test it, as I have the same problem on my Mac.

But perhaps we can leave this open anyway, as a reminder to properly fix this along the lines discussed in PR #99 ?

**Comment by fingolfin** [comment]
And yes, it seems to have fixed the issue for me! So a new release would be useful
